gravitee-apim-definition/gravitee-apim-definition-model/src/main/java/io/gravitee/definition/model/Api.java

```diff
@@ -241,6 +240,28 @@ public void setDefinitionContext(DefinitionContext definitionContext) {
         this.definitionContext = definitionContext;
     }
 
+    @JsonIgnore
+    public List<Plugin> getPlugins() {
+        return Stream
+            .of(
+                Optional
+                    .ofNullable(this.getResources())
+                    .map(r -> r.stream().map(Resource::getPlugins).flatMap(List::stream).collect(Collectors.toList()))
+                    .orElse(List.of()),
+                Optional
+                    .ofNullable(this.getFlows())
+                    .map(f -> f.stream().map(Flow::getPlugins).flatMap(List::stream).toList())
+                    .orElse(List.of()),
+                Optional
+                    .ofNullable(this.getPlans())
+                    .map(p -> p.stream().map(Plan::getPlugins).flatMap(List::stream).toList())
+                    .orElse(List.of()),
+                Optional.ofNullable(this.getServices()).map(Services::getPlugins).orElse(List.of())
+            )
+            .flatMap(List::stream)
+            .collect(Collectors.toList());
+    }
```

Comment:
Do you mean we need to override implementation in each Service implementation class ? For now I have a generic implem in Service.java but the type is `service` and id is `name`. This will not work because of the name of the features in the license, will it ?

Reply:
I was thinking about changing the Services (with an 's') to implement something specific because calling service.getPlugins() on each services will not work, unfortunately.
Probably something such as:

```java
public Set<Plugin> getPlugins() {
        final Set<Plugin> plugins = new HashSet<>();
        final EndpointDiscoveryService discoveryService = getDiscoveryService();
        
        if (discoveryService != null && discoveryService.isEnabled()) {
            plugins.add(new Plugin("service_discovery", discoveryService.getProvider()));
        }
   // .... same for HC and dynamic props
}
```

Reply:
Is the same result expected for api.services and api.proxy.endpointgroups.services ?

Reply:
Yes it is. It's just that some services are set for API (dynamic props) whereas others are set for endpoint groups (discovery, healthcheck) but the implementation can be unique.